server/src/test/java/org/elasticsearch/common/io/stream/AbstractStreamTests.java

```diff
@@ -256,12 +256,26 @@ public int hashCode() {
             StreamOutput::writeCollection,
             in -> in.readList(FooBar::new)
         );
+
+        runWriteReadOptionalCollectionTest(
+            () -> new FooBar(randomInt(), randomInt()),
+            StreamOutput::writeOptionalCollection,
+            in -> in.readOptionalList(FooBar::new)
+        );
     }
 
     public void testStringCollection() throws IOException {
         runWriteReadCollectionTest(() -> randomUnicodeOfLength(16), StreamOutput::writeStringCollection, StreamInput::readStringList);
     }
 
+    public void testOptionalStringCollection() throws IOException {
+        runWriteReadCollectionTest(
```

Comment:
Instead of relying on randomization, can we have explicit tests for the relevant cases? The data for the collection can be randomized, but we should explicitly test null and not null.